### PR TITLE
Document YAML data output that Tern produces

### PR DESCRIPTION
**Pull request by ForgetMe17**
Listing the all the keys of the YAML output and their description.

Works towards #175 

Signed-off-by: WangJL <hazard15020@gmail.com>

### CODE REVIEW

File: docs/yaml_output.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - `copyright`: copyright text
+    - `proj_url`: package source url
+    - `download_url`: package download url
+    - `origins`: a list of NoticeOrigin objects(notice_origin.py, expanded below)
+    - `checksum`: checksum as package property
+  - `origins`: list of NoticeOrigin objects (notice_origin.py, expended below)
+  - `tar_file`: the path to the layer filesystem tarball
+  - `created_by`: sometimes the metadata will contain a created_by
+      key containing the command that created the filesystem layer
+  - `import_image`: if the layer is imported from another image this is a pointer to that image. In Python terms it is just the name of the Image object or any object that uses this layer Based on how container image layers are created, this is usually the last layer of the image that was imported
```
Comment (nishakm) [inline docs/yaml_output.md]: IIRC nothing shows up in the YAML here. It's probably a bug we need to fix.

**Comment by nishakm** [comment]
@ForgetMe17 Thanks for this PR. It was a lot of work to summarize and this looks good to me. @rnjudge can you take a look as well and see if I am not missing anything?

File: docs/yaml_output.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# YAML Format Output
+
+You can get the results in a YAML file to be consumed by a downstream tool or script.
+
+`$ tern -l report -f yaml -i golang:1.12-alpine -o output.yaml`
+
+A yaml foramt output file starts with line comments indicating the version of Tern. It has a `image` key(image.py) corresponding to the analyzed image. Here are the subkeys and their description:
```
Comment (rnjudge) [inline docs/yaml_output.md]: small nit: `YAML` instead of `yaml and `format` instead of `foramt`

File: docs/yaml_output.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+- `config`: the image config metadata
```
Comment (rnjudge) [inline docs/yaml_output.md]: @nishakm do we want to list the subkeys of the `config` key similar to how we do with `packages` below? There are a few interesting ones like `architecture`, `created` and `os`.

File: docs/yaml_output.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  - `origins`: list of NoticeOrigin objects (notice_origin.py, expended below)
+  - `tar_file`: the path to the layer filesystem tarball
+  - `created_by`: sometimes the metadata will contain a created_by
+      key containing the command that created the filesystem layer
+  - `import_image`: if the layer is imported from another image this is a pointer to that image. In Python terms it is just the name of the Image object or any object that uses this layer Based on how container image layers are created, this is usually the last layer of the image that was imported
+  - `import_str`: The string from a build tool (like a Dockerfile) that created this layer by importing it from another image
+  - `files_analyzed`: whether the files in this layer are analyzed or not
+  - `analyzed_output`: the result of the file analysis
+
+`origins` is the origin of a notice, which is expanded as followed.
```
Comment (rnjudge) [inline docs/yaml_output.md]: small nit: `follows` instead of `followed`.

File: docs/yaml_output.md (hunk 3 above)
Comment (rnjudge) [reply docs/yaml_output.md]: On second thought, this might be a lot of work as there are lots of other subkeys with metadata as well :)

**Comment by rnjudge** [comment]
I requested just a few small changes. Otherwise, looks good! Thank you @ForgetMe17

File: docs/yaml_output.md (hunk 1 above)
Comment (ForgetMe17) [reply docs/yaml_output.md]: In the YAML output from the latest Release 1.0.1, I find theses keys under the `image`-`layers`-`analyzed_output`. For image with multiple layers, the analysis of each layer is under the key `analyzed_output`.

**Comment by ForgetMe17** [comment]
Thanks for your review!

**Comment by nishakm** [comment]
Looks good to me. @rnjudge?

**Comment by rnjudge** [comment]
> Looks good to me. @rnjudge?

LGTM!

**Pull request merged**